com.unity.multiplayer.mlapi/Runtime/Core/NetworkUpdateManager.cs

```diff
@@ -0,0 +1,324 @@
+using System;
+using System.Text;
+using UnityEngine;
+using UnityEngine.PlayerLoop;
+using UnityEngine.LowLevel;
+
+namespace MLAPI
+{
+    /// <summary>
+    /// Allows one to create their own network update engine
+    /// in the event they have specific processing needs etc.
+    /// </summary>
+    public interface INetworkUpdateEngine
+    {
+        void PreUpdate();
+        void PreUpdateRegister(Action updateAction);
+
+        void FixedUpdate();
+        void FixedUpdateRegister(Action updateAction);
+
+        void Update();
+        void UpdateRegister(Action updateAction);
+
+        void PostUpdate();
+        void PostUpdateRegister(Action updateAction);
+    }
+
+    /// <summary>
+    /// InternalNetworkUpdateEngine
+    /// Basic-Default version of an INetworkUpdateEngine implementation
+    /// This allows for anyone to come up with custom variations, which could include allowing multiple actions to be registered
+    /// </summary>
+    internal class InternalNetworkUpdateEngine : INetworkUpdateEngine
+    {
+        public Action PreUpdateAction;
+        public Action PostPreUpdateAction;
+        public Action FixedUpdateAction;
+        public Action UpdateAction;
+        public Action PostUpdateAction;
```

Comment:
I think we should have a comment block on top of this file which notes that this NetworkUpdateEngine (or rather NetworkLoopSystem) implementation will be revisited with another RFC and PR in the (very near) future.

Reply:
I agree with this, and will this to a new RPCQueue Revisions Jira Task.